src/test/java/com/puppycrawl/tools/checkstyle/grammars/AstRegressionTest.java

```diff
@@ -203,6 +203,12 @@ public void testImpossibleValid() throws Exception {
                 '4', '4', '.', '4', 'P', '4', 'D', ';');
     }
 
+    @Test
+    public void testJavadocPosition() throws Exception {
+        verifyAst(getPath("InputJavadocPositionAst.txt"),
+                getPath("InputJavadocPosition.java"), true);
```

Comment:
This is not a true test.
`AstRegressionTest` just calls `AstTreeStringPrinter.printFileAst` which just calls the Java grammar.
Only `AstTreeStringPrinter.printJavaAndJavadocTree` will parse javadoc comments.

This test should pass with or without your other changes.
I also don't believe this test, related to Javadoc, should be in the `AstRegressionTest` which is more concerned with Java grammar.